src/basic.rs

```diff
@@ -4,6 +4,189 @@ use std::{cmp::Ordering, unreachable};
+        Self {
+            rep: shuffle02(splat(f64::INFINITY), div_ru(x, y)),
+        }
+    }
+
+    #[inline]
+    pub(crate) fn div_p_n1(self, rhs: Self) -> Self {
+        // P / N1 => [b/d, a/c] = [-b/d; a/c] = [-b/d; -a/-c] = [-b; -a] ./ [d; -c]
+        let x = swap(self.rep); // [b; -a]
+        let x = neg0(x); // [-b; -a]
+        let y = swap(rhs.rep); // [d; -c]
+        Self { rep: div_ru(x, y) }
+    }
+
+    #[inline]
+    pub(crate) fn div_p_p0(self, rhs: Self) -> Self {
+        // P / P0 => [a/d, +∞] = [-a/d; +∞]
+        let x = self.rep; // [-a; b]
+        let y = swap(rhs.rep); // [d; -c]
+        Self {
+            rep: shuffle02(div_ru(x, y), splat(f64::INFINITY)),
+        }
+    }
+
+    #[inline]
+    pub(crate) fn div_p_p1(self, rhs: Self) -> Self {
+        // P / P1 => [a/d, b/c] = [-a/d; b/c] = [-a; b] ./ [d; c]
+        let x = self.rep; // [-a; b]
+        let y = neg0(rhs.rep); // [c; d]
+        let y = swap(y); // [d; c]
+        Self { rep: div_ru(x, y) }
+    }
+
+    /// Return the extended division of `numerator`$\setdiv \self$
+    /// (also called *two-output division* in the IEEE 1788 standard).
+    ///
+    /// The set-division of two intervals $𝒙$ and $𝒚$ is defined as
+    /// $$𝒙 \setdiv 𝒚 := \set{z ∈ \R ∣ ∃y ∈ 𝒚,\ zy ∈ 𝒙}.$$
+    /// Let us distinguish several cases.
+    /// - If $𝒙$ or $𝒚$ is empty, $𝒙 \setdiv 𝒚 = ∅$.  Thus if `self`
+    ///   or `numerator` is empty, this function returns `[EMPTY, EMPTY]`.
+    /// - If $0 ∉ 𝒚$, $𝒙 \setdiv 𝒚$ is a single interval $𝒛$ which
+    ///   coincides with the standard interval division $𝒙 / 𝒚$.  Then
+    ///   $𝒚$`.mul_rev_to_pair`($𝒙$) returns `[z, EMPTY]` where `z` is
+    ///   an encosure of $𝒛$.
+    /// - If $0 ∈ 𝒚$ and $0 ∉ 𝒙$, $𝒙 \setdiv 𝒚$ is a made of two
+    ///   intervals $𝒛₁ ∪ 𝒛₂$.  The standard division $𝒙 / 𝒚$ returns
+    ///   the interval enclosure of the result, namely $ℝ$.  Here,
+    ///   $𝒚$`.mul_rev_to_pair`($𝒙$) returns $[𝒛₁, 𝒛₂]$ ordered such
+    ///   that $𝒛₁ < 𝒛₂$.
+    /// - If $0 ∈ 𝒚$ and $0 ∈ 𝒙$, $𝒙 \setdiv 𝒚 = ℝ$.  Accordingly,
+    ///   $𝒚$`.mul_rev_to_pair`($𝒙$) return `[ENTIRE, EMPTY]`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use inari::{Interval as I, const_interval as c};
+    /// let zero = c!(0., 0.);
+    /// assert_eq!(zero.mul_rev_to_pair(c!(1., 2.)), [I::EMPTY; 2]);
+    /// assert_eq!(zero.mul_rev_to_pair(c!(0., 2.)), [I::ENTIRE, I::EMPTY]);
+    /// let x = c!(1., 2.);
+    /// assert_eq!(c!(1., 1.).mul_rev_to_pair(x), [x, I::EMPTY]);
+    /// assert_eq!(c!(1., f64::INFINITY).mul_rev_to_pair(c!(1., 1.)),
+    ///            [c!(0., 1.), I::EMPTY]);
+    /// assert_eq!(c!(-1., 1.).mul_rev_to_pair(c!(1., 2.)),
+    ///            [c!(f64::NEG_INFINITY, -1.), c!(1., f64::INFINITY)]);
+    /// assert_eq!(c!(-1., 1.).mul_rev_to_pair(zero), [I::ENTIRE, I::EMPTY]);
+    /// ```
+    #[must_use]
+    pub fn mul_rev_to_pair(self, numerator: Self) -> [Self; 2] {
```

Comment:
Just a thought. How about renaming `numerator` to `rhs`, as it is the right-hand side of the multiplication?

Reply:
Also, at least for me, having rust-analyser show `numerator` helps to check that I did not swap the two arguments...